ddl/column.go

```diff
@@ -708,6 +708,11 @@ func needChangeColumnData(oldCol, newCol *model.ColumnInfo) bool {
 			return isElemsChangedToModifyColumn(oldCol.Elems, newCol.Elems)
 		case mysql.TypeTiny, mysql.TypeShort, mysql.TypeInt24, mysql.TypeLong, mysql.TypeLonglong:
 			return toUnsigned != originUnsigned
+		case mysql.TypeString:
```

Comment:
Maybe it isn't limited to TypeString?
should the below be right?
```
default:
    types.IsString(colDef.Tp.Tp) && mysql.HasBinaryFlag(colDef.Tp.Flag)
    ...
```

Reply:
sorry, I'm wrong, I found that MySQL won't change the length for blob/text/char/varchar, seems only for binary(x)?

Reply:
I think you can use this condition for if statement: 
refer it in castvalue function:
if tp.Tp == mysql.TypeString && IsBinaryStr(tp) && len(s) < flen && padZero {

Reply:
Yes, `IsBinaryStr(tp)` seems more clear in semantic.